### ISSUE TITLE
The weights argument in classif.log_reg are not observation weights

### ISSUE THREAD

**Issue by salauer** [mlr-org/mlr3learners#177]
I ran into this (in a bad way) and I don't have a good solution to it.

The `weights` argument of the `glm()` function does not indicate observation weights as we are used to for other learners. From the `glm` documentation (bold emphasis is mine):

>Non-`NULL` `weights` can be used to indicate that different observations have different dispersions (with the values in `weights` being inversely proportional to the dispersions); or equivalently, when the elements of `weights` are positive integers *w_i*, that each response *y_i* is the mean of *w_i* unit-weight observations. **For a binomial GLM prior weights are used to give the number of trials when the response is the proportion of successes**: they would rarely be used for a Poisson GLM.

Therefore, when using the learner `classif.log_reg` (which calls `glm(..., family="binomial")`), using weights isn't changing the observation weights but actually changing the outcome by giving it a denominator other than 1 (see this SO post for more detail: https://stats.stackexchange.com/a/386913/137894). And these are only "prior weights" so even those change in the fitting process.

Here's a minimal example:
```
## create some data
set.seed(1)
eg_df <- data.frame(group=as.factor(rbinom(10,1,0.8)),
                    x = rnorm(10))
## there are 7 1s and 3 0s so weight the 0s up
eg_df$wts <- ifelse(eg_df$group == 1, 1, 7/3)

eg_glm <- glm(group ~ x, family="binomial", data=eg_df, weights=wts)
## weights aren't as specified
unname(eg_glm$weights)
> [1] 0.081512108 0.248465132 0.226602079 0.568258751 0.164051645 0.227390656 0.583221100 0.109680339
  [9] 0.006920066 0.162130509

## same in mlr3
learner <- lrn("classif.log_reg")
task <- TaskClassif$new(id = "eg", backend = eg_df, target = "group")
task$set_col_roles(cols = "wts", roles = "weight")
learner$train(task)
unname(learner$model$weights)
> [1] 0.081512108 0.248465132 0.226602079 0.568258751 0.164051645 0.227390656 0.583221100 0.109680339
  [9] 0.006920066 0.162130509
```

Again, I'm not quite sure how to solve this (I'll look at a few other packages), but we probably don't want the learner treating the weights like this (lines 62-64 of `R/LearnerClassifLogReg.R`):
```
if ("weights" %in% task$properties) {
        pars = mlr3misc::insert_named(pars, list(weights = task$weights$weight))
      }
```

**Comment by mllg** [comment]
@berndbischl your thoughts on this?

**Comment by mllg** [comment]
Pinging @berndbischl again :)

**Comment by be-marc** [comment]
@salauer Thank you for pointing this out. We have removed the option to use weights. Closed by #247.
